bika/lims/workflow/analysis/events.py

```diff
@@ -187,8 +201,11 @@ def after_attach(obj):
     _reindex_request(obj)
 
 
-def _reindex_request(obj):
-    if IRoutineAnalysis.providedBy(obj):
-        request = obj.getRequest()
-        if request:
-            request.reindexObject()
+def _reindex_request(obj, idxs=None):
+    if not IRequestAnalysis.providedBy(obj):
```

Comment:
What's the difference between `IRequestAnalysis` and `IAnalysis`?

Reply:
`IRequestAnalysis` interface is implemented by those analyses that belong, directly or indirectly, to an `AnalysisRequest`. At the moment, `Analysis` (directly), `DuplicateAnalysis` (indirectly). As such, thsese objects implement the function `getRequest` (and related).

`ReferenceAnalysis` objects are not associated to `AnalysisRequest`, rather their parent is directly a `Worksheet`, so they don't implement the function `getRequest` and related.

Reply:
Thanks for the explanation Jordi